src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -59,13 +61,60 @@ public Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, Invocati
 
                     RaiseDocumentDiagnosticsIfNeeded(document, stateSet, kind, result.OldItems, result.Items);
                 }
+
+                await ReportAnalyzerPerformanceAsync(document, analyzerDriverOpt, cancellationToken).ConfigureAwait(false);
             }
             catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
             {
                 throw ExceptionUtilities.Unreachable;
             }
         }
 
+        private async Task ReportAnalyzerPerformanceAsync(Document document, CompilationWithAnalyzers analyzerDriverOpt, CancellationToken cancellationToken)
+        {
+            if (analyzerDriverOpt == null)
+            {
+                return;
+            }
+
+            var client = await document.Project.Solution.Workspace.TryGetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
+            if (client == null)
+            {
+                // no remote support
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using (var pooledObject = SharedPools.Default<Dictionary<DiagnosticAnalyzer, AnalyzerTelemetryInfo>>().GetPooledObject())
+            {
+                var containsData = false;
+                foreach (var analyzer in analyzerDriverOpt.Analyzers)
+                {
+                    var telemetryInfo = await analyzerDriverOpt.GetAnalyzerTelemetryInfoAsync(analyzer, cancellationToken).ConfigureAwait(false);
+                    if (!containsData && telemetryInfo.ExecutionTime.Ticks > 0)
+                    {
+                        // this is unfortunate tweak due to how GetAnalyzerTelemetryInfoAsync works when analyzers are asked
+                        // one by one rather than in bulk.
```

Comment:
This data is also exposed as an immutable dictionary if that helps avoid this hack: [http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalysisResult.cs,56](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalysisResult.cs,56)

Reply:
@mavasani can I use AnalysisResult when I am using same compilatinWithAnalyzer and calling it each analyzer rather than bulk run them?

Reply:
talked with @mavasani offline. I can't use that API for this.